### PR TITLE
Add a flag to IO to ignore ACG1/2/3 in manual mode.

### PR DESCRIPTION
**Pull request by bartosz-wawrzacz**
This doesn't change the existing functionality in any way, just adds a new option for the IO that you can use in the described situation.

TL;DR: let's you use a `nofmu` option when starting px4io to make it only mix ACG0 when in MANUAL vel OVERRIDE mode.

Backstory:
Used if you don't have enough RC channels available to tie every control surface to one channel but still want the possibility to control them directly from the FMU.

We have a twin-motor V-tail UAV, which gives you 6 actuators. Together with two mode switches (MODE_SW, POSCTL_SW), flaps switch and an aditional AUX switch we need for our custom controller, that makes 10 channels that we would need, but only have 8 on the DX8.
We don't want to do any mixing on the RC remote, because then you have to de-mix everything to get the raw commands for the controller.

The RC channels setup is as follows and this goes into the mixer as Actuator Control Group 0:
`roll, pitch, throttle, yaw, flaps, mode_sw, posctl_sw, aux`

Then all the necessary mixing is done in the IO mixer. Eq. aileron commands are computed as:
`AIL_L = roll (with applied differential) + flaps`
`AIL_R = roll (with applied differential) - flaps`
Also the motors are both feeded with the same `throttle` value.

This is fine for manual control but doesn't give you full authority over the actuators in any autopilot-assisted modes.

The solution to this to publish zeros on ACG0 and the actual control values on ACG1 by the FMU.
There is a drawback to this approach, as the mixers get a bit bigger:
`AIL_L = roll + differential (ACG0) + flaps (ACG0) + ail_l (ACG1)`
`AIL_R = roll + differential (ACG0) - flaps (ACG0) + ail_r (ACG1)`
But this looks like the only option we have in this situation.

Now the actual reason for adding the flag. For this method to work, FMU has to publish zeros on ACG1 when in manual mode or otherwise we will get double input to the mixer (ACG0 coming from the RC, ACG1 from FMU).
But when switching modes, there is a slight delay between IO and FMU switching, which introduces some awful stuttering.
To visualise:
- manual mode - RC values on ACG0, zeros on ACG1
- switching to FMU - IO switches immediately
- zeros on ACG0, zeros on ACG1 - actuators jump!
- FMU realises the mode changed
- FMU mode - zeros on ACG0, control values on ACG1
  Same problem when switching the other way.

The solution is to mix in only the ACG0 values when in manual mode. This way the FMU can simply publish it's output on ACG1 all the time and no stuttering will happen while switching.

One caveat:
The mixer [runs the `mixer_callback` function](https://github.com/bartosz-wawrzacz/Firmware/blob/ae1fe9d3b43115206fea923f4a00b940317ea626/src/modules/systemlib/mixer/mixer_simple.cpp#L293) which fetches the input value from the appropriate IO register:
https://github.com/bartosz-wawrzacz/Firmware/blob/00e0a352908d05fb26c7ac919d9a3fab86a16cef/src/modules/px4iofirmware/mixer.cpp#L345
The if statement here returns `false` for ACG1 inputs, as there is no RC input in this group. As a result, the callback returns -1 and doesn't modify the `control` variable. The [small fix](https://github.com/bartosz-wawrzacz/Firmware/commit/ae1fe9d3b43115206fea923f4a00b940317ea626#diff-da07bbd8970e171133193eb7c155906fR291) in `mixer_simple.cpp` solves this.

### CODE REVIEW

**Comment by bartosz-wawrzacz** [comment]
@LorenzMeier Could you maybe have a quick look at this and tell if it's viable?
Or perhaps there is a better way to solve the described problem?

Actually, if you have a 10-channel RC system, where do the CH9 and CH10 values end up?

**Comment by LorenzMeier** [comment]
I've looked into this but will need more time to really understand the issue and the solution. Thanks!

**Comment by TSC21** [comment]
@LorenzMeier is this still to review?

**Comment by LorenzMeier** [comment]
It's alright but needs a rebase. I get now what it's trying to do.
